src/include/btree_inline.h

```diff
@@ -1793,41 +1793,20 @@ static inline bool
 __wt_page_evict_retry(WT_SESSION_IMPL *session, WT_PAGE *page)
 {
     WT_PAGE_MODIFY *mod;
-    WT_TXN_GLOBAL *txn_global;
-    wt_timestamp_t pinned_ts;
 
-    txn_global = &S2C(session)->txn_global;
-
-    /*
-     * If the page hasn't been through one round of update/restore, give it a try.
-     */
-    if ((mod = page->modify) == NULL || !FLD_ISSET(mod->restore_state, WT_PAGE_RS_RESTORED))
-        return (true);
+    mod = page->modify;
 
-    /*
-     * Retry if a reasonable amount of eviction time has passed, the choice of 5 eviction passes as
-     * a reasonable amount of time is currently pretty arbitrary.
-     */
-    if (__wt_cache_aggressive(session) ||
-      mod->last_evict_pass_gen + 5 < S2C(session)->cache->evict_pass_gen)
+    if (mod == NULL)
         return (true);
 
-    /* Retry if the global transaction state has moved forward. */
-    if (txn_global->current == txn_global->oldest_id ||
-      mod->last_eviction_id != __wt_txn_oldest_id(session))
```

Comment:
Here it says if last_eviction_id is the oldest, we will not retry. This is blocking the eviction to add the page to the queue.

Reply:
My understanding is the same as Chenhao's. Before durable history eviction was used to evict only globally visible content, later we modified it such that eviction can have its own snapshot and it could be later than a checkpoint.

Reply:
If we remove this check, in a situation that the oldest id doesn't move but there are also not many new transactions committing, we will inefficiently add those pages with uncommitted changes to the eviction queue. This will decrease the eviction efficiency. I'm struggling to think of if it is possible to only put pages that have recent commits to the queue. I think it is hard because in the commit code, we only change the states of the updates but not anything on the page level. Therefore, it is costly for us to trace what is the latest committed transaction on each page as we need to do a full search for each committed update in order to update the latest committed transaction on that page, which is extremely costly.